vllm_spyre/v1/core/scheduler.py

```diff
@@ -341,4 +362,31 @@ def check_batch_tkv_limit(self, request, tkv) -> bool:
                 # decrease batch_size by 1 as the current request finished
                 batch_size -= 1
 
-        return max_batch_tkv <= int(self.max_batch_tkv_limit)
+        return_value = max_batch_tkv <= int(max_batch_tkv_limit)
+
+        if outer_key in cache:
+            # decode batch has not changed
+            if inner_key not in cache[outer_key]:
+                # adding new request to present decode batch
+                cache[outer_key][inner_key] = (-float("inf"), float("inf"))
+        else:
+            # decode batch has changed, empty the cache in place
+            cache.clear()
+            cache[outer_key] = {inner_key: (-float("inf"), float("inf"))}
```

Comment:
NIT: possible micro-optimization, `float("inf")` makes think it will parse, while we have already a constant for that.
```suggestion
            cache[outer_key] = {inner_key: (-math.inf, math.inf)}
```